mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java

```diff
@@ -58,19 +57,20 @@ public void init() throws javax.mail.MessagingException {
 
     @Override
     public Collection<MailAddress> match(Mail mail) throws javax.mail.MessagingException {
-        Optional<String> mimeTypes = getMimeTypeFromMessage(mail.getMessage());
-
-        return mimeTypes.filter(acceptedContentTypes::contains)
-                .map(any -> mail.getRecipients())
-                .orElse(ImmutableList.of());
+        return StreamUtils.ofNullable(mail.getMessage().getHeader(CONTENT_TYPE))
+            .flatMap(this::getMimeType)
+            .filter(acceptedContentTypes::contains)
+            .findAny()
```

Comment:
why are you using streams while you just need an Optional?

Reply:
because there might be several content type header